src/test/java/com/amazonaws/services/dynamodbv2/GetLocksByPartitionKeyTest.java

```diff
@@ -0,0 +1,175 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.Test;
+
+import static java.util.stream.Collectors.toList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class GetLocksByPartitionKeyTest extends InMemoryLockClientTester {
+  /**
+   * <p>1 MB</p>
+   * <a href="http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/QueryAndScan.html#Pagination">
+   * Dynamo DB Max Page Size Documentation
+   * </a>
+   */
+  private static final int DYNAMODB_MAX_PAGE_SIZE_IN_BYTES = 1 << 20;
+
+  /**
+   * <p>400 KB</p>
+   * <a href="http://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.BatchOperations">
+   * Dynamo DB Max Item Size Documentation
+   * </a>
+   */
+  private static final int DYNAMODB_MAX_ITEM_SIZE_IN_BYTES = 400 * (1 << 10);
+
+  @Test
+  public void getLocksFromDynamoDB_whenNoLocks_shouldReturnEmpty() throws LockNotGrantedException {
+    final boolean deleteOnRelease = false;
+    final List<LockItem>
+        locksWithPartitionKey = this.lockClientForRangeKeyTable.getLocksByPartitionKey("test", deleteOnRelease).collect(toList());
+
+    assertEquals(Collections.emptyList(), locksWithPartitionKey);
+  }
+
+  @Test
+  public void getLocksFromDynamoDB_whenSingleLock_shouldReturnSingleLock() throws LockNotGrantedException, InterruptedException {
+    final String partitionKey = "Test 1";
+    final String sortKey = "1";
+    final AcquireLockOptions options = AcquireLockOptions.builder(partitionKey)
+        .withSortKey(sortKey).withData(ByteBuffer.wrap(TEST_DATA.getBytes())).withDeleteLockOnRelease(true).build();
+
+    final LockItem singleLock = this.lockClientForRangeKeyTable.acquireLock(options);
+
+    final boolean deleteOnRelease = false;
+    List<LockItem> locksWithParitionKey = this.lockClientForRangeKeyTable.getLocksByPartitionKey(partitionKey, deleteOnRelease).collect(toList());
+
+    assertEquals(1, locksWithParitionKey.size());
+
+    final LockItem retrievedLock = locksWithParitionKey.get(0);
+    assertEquals(singleLock.getPartitionKey(), retrievedLock.getPartitionKey());
+    assertEquals(singleLock.getSortKey(), retrievedLock.getSortKey());
+    assertTrue(
+        Arrays.equals(getBytes(singleLock.getData().get()), getBytes(retrievedLock.getData().get())));
+    assertEquals(singleLock.getRecordVersionNumber(), retrievedLock.getRecordVersionNumber());
+
+    this.lockClientForRangeKeyTable.getLock(partitionKey, Optional.of(sortKey)).get().close();
+
+    locksWithParitionKey = this.lockClientForRangeKeyTable.getLocksByPartitionKey(partitionKey, deleteOnRelease).collect(toList());
+    assertEquals(Collections.emptyList(), locksWithParitionKey);
+  }
+
+  @Test
+  public void getLocksFromDynamoDB_whenMultipleLocks_shouldReturnMultipleLocks() throws LockNotGrantedException, InterruptedException, IOException {
```

Comment:
Let's add a similar test where there are locks with different partition key value which should not be returned